x-pack/filebeat/module/gcp/audit/config/pipeline.js

```diff
@@ -63,6 +63,35 @@ function Audit(keep_original_message) {
         fail_on_error: false,
     });
 
+    var setOrchestratorMetadata = function(evt) {
+          if (evt.Get("json.resource.type") === "k8s_cluster") {
+            evt.Put("orchestrator.type", "kubernetes");
+            // Dissect to extract the api_version
+            var dissect_processor = new processor.Dissect({
+              "tokenizer": "%{}/%{orchestrator.api_version}/%{}",
```

Comment:
based on ur comment about the varying formats of the API version, I would move this to a grok statement.  Then you could do something like below.  This would need to be done on ES pipeline side.
```
- grok:
     field:
     patterns:
        - '/%{API_VERSION:orchestrator.api_version}'
     pattern_definitions:
        API_VERSION: (v\d+([a-z]+)?(\d+)?)
```

Reply:
See the below comment for a fully built out grok processor to get API version and resource types/names